payment_processors/braintree_processor/models/payment_processors.py

```diff
@@ -44,43 +43,39 @@ def refund_transaction(self, transaction, payment_method=None):
     def void_transaction(self, transaction, payment_method=None):
         pass
 
-    def _update_payment_method(self, payment_method, result_payment_method):
+    def _update_payment_method(self, payment_method, result_details, type):
         """
         :param payment_method: A BraintreePaymentMethod.
         :param result_payment_method: A payment method from a braintreeSDK
                                       result(response).
         :description: Updates a given payment method's data with data from a
                       braintreeSDK result payment method.
         """
-
         payment_method_details = {
-            'type': result_payment_method.__class__.__name__
+            'type': type,
+            'image_url': result_details.image_url,
+            'updated_at': timezone.now().isoformat()
         }
 
-        if payment_method_details['type'] == payment_method.Type.PayPal:
-            payment_method_details['email'] = result_payment_method.email
-        elif payment_method_details['type'] == payment_method.Type.CreditCard:
+        if type == payment_method.Types.PayPal:
```

Comment:
`type` is a built-in function in Python. Use `kind`,  another synonym or `payment_type`.